src/layouts/primary-layout.js

```diff
@@ -20,76 +20,52 @@ import ProfilePage from '../pages/profile-page'
 import SelectionProcessPage from '../pages/selection-process-page'
 import TracksGuidePage from '../pages/tracks-guide-page'
 import PresentationLayout from './presentation-layout'
-import { getAllFromSummit } from '../actions/base-actions';
+import {getSelectionPlan} from "../actions/base-actions";
 
-class PrimaryLayout extends React.Component {
+const PrimaryLayout = ({location, summit, speaker, member, match, selectionPlan, getSelectionPlan}) => {
+    const loggedUser = (speaker && speaker.id) ? speaker : member;
+    const selectionPlanIdParam = parseInt(match.params.selection_plan_id);
```

Comment:
could be useMemo here and then add it as dependency on useEffect?

Reply:
useMemo is for expensive calculations, this is just a parse, I don't think we need to memoize